src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts

```diff
@@ -2656,33 +2657,44 @@ export class InstallVSIXAction extends Action {
 		label = InstallVSIXAction.LABEL,
 		@IExtensionsWorkbenchService private extensionsWorkbenchService: IExtensionsWorkbenchService,
 		@INotificationService private notificationService: INotificationService,
-		@IWindowService private windowService: IWindowService
+		@IWindowService private windowService: IWindowService,
+		// {{SQL CARBON EDIT}}
+		@IConfigurationService private configurationService: IConfigurationService
 	) {
 		super(id, label, 'extension-action install-vsix', true);
 	}
 
 	run(): TPromise<any> {
-		return this.windowService.showOpenDialog({
-			title: localize('installFromVSIX', "Install from VSIX"),
-			filters: [{ name: 'VSIX Extensions', extensions: ['vsix'] }],
-			properties: ['openFile'],
-			buttonLabel: mnemonicButtonLabel(localize({ key: 'installButton', comment: ['&& denotes a mnemonic'] }, "&&Install"))
-		}).then(result => {
-			if (!result) {
-				return TPromise.as(null);
-			}
+		// {{SQL CARBON EDIT}}
+		let extensionPolicy = this.configurationService.getValue<string>(ExtensionsPolicyKey);
+		if (extensionPolicy !== ExtensionsPolicy.allowNone) {
```

Comment:
Here the code checks if the extension policy is not `allowNone` but in the next file it checks if the extension policy is `allowAll`. Should "Install from VSIX" be enabled when the policy is set to Microsoft only? Those should all get auto installed so I think we can completely disable "Install from VSIX" in that case right?

Reply:
"Install from VSIX" is only enabled for "allowAll". The initial check for "allowNone" is to suppress the dialog to pick an extension if extensions aren't allowed at all.

Reply:
This code is in the "install from vsix" action though